### PR TITLE
Make indexPath a File.

### PR DESCRIPTION
**Pull request by dozed**
I needed to write the digest index to `(resourceManaged in Compile)`. There is already `indexWriter` but `indexPath` is limited to a `String` which points to a file inside `webTarget`.

This PR changes `indexPath` to `indexFile`.

``` scala
DigestKeys.indexWriter := {
  index => index.asJson.compact
},

DigestKeys.indexFile := Some((resourceManaged in Compile).value / "assets.index.json")
```

And:

``` scala
DigestKeys.indexWriter ~= {
  writer => index => s"var versioned = ${writer(index)};"
},

DigestKeys.indexFile := Some(WebKeys.webTarget.value / "javascripts/versioned.js")
```

### CODE REVIEW

**Comment by dozed** [comment]
@pvlugter any thoughts on this?

File: src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala [hunk 1]
```diff
@@ -104,24 +104,27 @@ object SbtDigest extends AutoPlugin {
...
+      indexFileOpt: Option[File],
       indexWriter: Map[String, String] => String
     ): Seq[PathMapping] = {
       val processed = process(mappings, algorithms, include, exclude, targetDir)
-      val digested = indexPathOpt.fold(processed) { indexPath =>
+      val digested = indexFileOpt.fold(processed) { indexFile =>
         // generate an index file and add to digested outputs
         // replace any existing file at the index path
-        val filtered = processed filterNot (_.originalPath == indexPath)
+        val filtered = processed filterNot (_.originalPath == indexFile.toString)
```
Comment (pvlugter) [inline src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala]: Is `indexFile.toString` the same as the previous `indexPath`? To be the same, won't it need to be a path relative to the target directory?

File: src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala [hunk 2]
```diff
@@ -104,24 +104,27 @@ object SbtDigest extends AutoPlugin {
...
             versioned.headOption map { v => normalizePath(v.originalPath) -> normalizePath(v.mapping.path) }
           case _ => None
         }).toMap
         val indexContent = indexWriter(indexMap)
-        val indexFile = targetDir / indexPath
         IO.write(indexFile, indexContent)
-        val digestedIndex = process(Seq(indexFile -> indexPath), algorithms, include, exclude, targetDir)
-        filtered ++ digestedIndex
+
+        val digestedIndex = IO.relativize(targetDir, indexFile) map { indexPath =>
```
Comment (pvlugter) [inline src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala]: What if `indexFile` is not relative to `targetDir`? Does it matter?

File: src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala [hunk 3]
```diff
@@ -14,7 +14,7 @@ object Import {
 
   object DigestKeys {
     val algorithms = SettingKey[Seq[String]]("digest-algorithms", "Types of checksum files to generate.")
-    val indexPath = TaskKey[Option[String]]("digest-index-path", "Path to the generated asset index file.")
+    val indexFile = TaskKey[Option[File]]("digest-index-file", "Generated asset index file.")
```
Comment (pvlugter) [inline src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala]: For compatibility, could we still have `indexPath` and create the `indexFile` from it, if it's not explicitly defined?

File: src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala (hunk 2 above)
Comment (pvlugter) [inline src/main/scala/com/typesafe/sbt/digest/SbtDigest.scala]: Okay, so there's no need to digest as it's not included?

**Comment by pvlugter** [comment]
This looks reasonable, and good to see the test that shows it works for you. I think it would be good to keep a compatible `indexPath` for anyone that was using it. And then have `indexFile` based on `indexPath` by default, but also overridable. Does that work for you?

**Comment by jroper** [comment]
Closing due to lack of activitiy.

**Pull request closed without merging**